src/main/webapp/src/app/nav-bar/nav-bar.component.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { ComponentFixture, TestBed, inject, async } from '@angular/core/testing';
+import { NavBarComponent } from './nav-bar.component';
+import { Location } from '@angular/common';
+import { By } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { RouterTestingModule } from '@angular/router/testing';
+import { routes } from '../app-routing.module'
+
+describe('NavBarComponent', () => {
+ 
+  let component: NavBarComponent;
+  let fixture: ComponentFixture<NavBarComponent>;
+  let location: Location;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ RouterTestingModule ],
+      declarations: [ NavBarComponent ],
+      providers: [
+        NavBarComponent
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        CommonModule,
```

Comment:
I don't really know much about it, but is CommonModule required here? I checked this link: https://angular.io/api/common/CommonModule ; and I'm not sure whether you use any directives or pipes here.

Reply:
removed it